parlai/agents/bart/modules.py

```diff
@@ -27,26 +26,37 @@ def output(self, tensor: torch.Tensor) -> torch.Tensor:
         output = F.linear(tensor, self.embeddings.weight)
         return output
 
-    def decode_forced(
-        self, encoder_states: Tuple[Any, ...], ys: torch.LongTensor
-    ) -> Tuple[torch.FloatTensor, torch.LongTensor]:
+    def _get_initial_forced_decoder_input(self, bsz: int, inputs: torch.LongTensor):
         """
-        Decode with a fixed, true sequence, computing loss.
+        Return initial input to the decoder.
+
+        Override TGA._get_initial_forced_decoder_input to seed EOS BOS.
 
-        Overriding `TGM.decode_forced` to bypass assertion that BOS is not present, and
-        additionally insert EOS as first token
+        :param bsz:
+            batchsize
+        :param inputs:
+            inputs to decode
+
+        :return initial_input:
+            initial input for the decoder.
         """
-        bsz = ys.size(0)
-        seqlen = ys.size(1)
-        inputs = ys.narrow(1, 0, seqlen - 1)
-        inputs = torch.cat(
-            [
-                torch.LongTensor([self.END_IDX]).detach().expand(bsz, 1).to(inputs),
-                inputs,
-            ],
-            1,
+        tens = (
+            torch.LongTensor([self.END_IDX, self.START_IDX])
```

Comment:
Hmm we could abstract out this intro sequence (`[self.END_IDX, self.START_IDX]`), which shows up at least twice

Reply:
I'm not sure we can, it shows up in separate classes (`BartAgent` vs. `BartModel`) and is dependent on the set attrs for `END_IDX` and `START_IDX` which we do not know beforehand